java-checks/src/test/files/checks/security/LDAPDeserializationCheck.java

```diff
@@ -0,0 +1,33 @@
+import javax.naming.directory.SearchControls;
+import java.util.Date;
+import java.util.Properties;
+
+class S4434 {
+
+  void nonCompliantCreate(int scope, long countLimit, int timeLimit, String[] attributes, boolean deref) {
+    SearchControls ctrl = new SearchControls(scope, countLimit, timeLimit, attributes, true, deref); // Noncompliant {{Disable object deserialization.}}
+  }
+
+  void nonCompliantSet() {
+    SearchControls ctrl = new SearchControls();
+    ctrl.setReturningObjFlag(true); // Noncompliant
+  }
+
+  void compliantCreate(int scope, long countLimit, int timeLimit, String[] attributes, boolean deref) {
+    new SearchControls(scope, countLimit, timeLimit, attributes, false, deref);
```

Comment:
The only difference with the first test case should be what triggers a different result.